src/components/menu/menu-interim-element.js

```diff
@@ -36,44 +36,112 @@ function MenuProvider($$interimElementProvider) {
+           $animate.leave(options.backdrop);
+         }
+         if (options.disableParentScroll) {
+           options.restoreScroll();
+         }
+       }
+     }
+
+    /**
+     * Boilerplate interimElement onRemove function
+     * Handles removing the menu from the DOM, cleaning up the element
+     * and removing various listeners
+     */
+    function onRemove(scope, element, opts) {
+      opts.cleanupInteraction();
+      opts.cleanupResizing();
+
+      return $animateCss(element, { addClass : 'md-leave' })
+        .start()
+        .then(function() {
+          element.removeClass('md-active');
+          opts.hideBackdrop();
+
+          detachElement(element, opts);
+          opts.alreadyOpen = false;
+        });
+    }
+
+    /**
      * Boilerplate interimElement onShow function
-     * Handles inserting the menu into the DOM, positioning it, and wiring up
-     * various interaction events
+     * Handles inserting the menu into the DOM, positioning it, and wiring up various interaction events
      */
     function onShow(scope, element, opts) {
-
-      // Sanitize and set defaults on opts
-      buildOpts(opts);
+      sanitizeAndConfigure(opts);
 
       // Wire up theming on our menu element
       $mdTheming.inherit(opts.menuContentEl, opts.target);
 
       // Register various listeners to move menu on resize/orientation change
-      handleResizing();
+      opts.cleanupResizing = activateResizing();
+      opts.hideBackdrop = showBackdrop(scope,element,opts);
 
-      // Disable scrolling
-      if (opts.disableParentScroll) {
-        opts.restoreScroll = $mdUtil.disableScrollAround(opts.element);
-      }
+      // Return the promise for when our menu is done animating in
+      return showMenu()
+        .then( function(response) {
+          opts.alreadyOpen = true;
+          opts.cleanupInteraction = activateInteraction();
+          return response;
+        });
 
-      if (opts.backdrop) {
-        $mdTheming.inherit(opts.backdrop, opts.parent);
-        opts.parent.append(opts.backdrop);
-      }
-      showMenu();
+      /**
+       * Place the menu into the DOM and call positioning related functions
+       */
+      function showMenu() {
+        opts.parent.append(element);
 
-      // Return the promise for when our menu is done animating in
-      return animator
-          .waitTransitionEnd(element, {timeout: 370})
-          .finally( function(response) {
-            opts.cleanupInteraction = activateInteraction();
-            return response;
-          });
+        return $q(function(resolve){
+
+          $animateCss(element, { removeClass: 'md-leave', duration:0 })
+            .start()
+            .then(function(){
+              var position = calculateMenuPosition(element, opts);
+
+              $animateCss(element, {
+                addClass : 'md-active',
+                from : animator.toCss( position ),
+                to : animator.toCss( { transform : 'scale(1.0,1.0)' })
```

Comment:
Remove spaces from inside object literal (here and elsewhere). E.g.,
`animator.toCss({ transform : 'scale(1.0,1.0)'})`